### PR TITLE
WIP: Get things working on v0.5

### LINKED ISSUE #209: Tests fail on v0.5
```
Error During Test
  Test threw an exception of type UndefVarError
  Expression: contains(sprint(show,complement(IntSet())),"...,")
  UndefVarError: complement not defined
   in include_from_node1(::String) at ./loading.jl:426
   in macro expansion; at /Users/ranjan/.julia/v0.5/DataStructures/test/runtests.jl:30 [inlined]
   in anonymous at ./<missing>:?
   in include_from_node1(::String) at ./loading.jl:426
   in process_options(::Base.JLOptions) at ./client.jl:262
   in _start() at ./client.jl:318
ERROR: LoadError: LoadError: There was an error during testing
 in record(::Base.Test.FallbackTestSet, ::Base.Test.Error) at ./test.jl:397
 in do_test(::Base.Test.Threw, ::Expr) at ./test.jl:281
 in include_from_node1(::String) at ./loading.jl:426
 in macro expansion; at /Users/ranjan/.julia/v0.5/DataStructures/test/runtests.jl:30 [inlined]
 in anonymous at ./<missing>:?
 in include_from_node1(::String) at ./loading.jl:426
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /Users/ranjan/.julia/v0.5/DataStructures/test/test_int_set.jl, in expression starting on line 17
while loading /Users/ranjan/.julia/v0.5/DataStructures/test/runtests.jl, in expression starting on line 27
```

### PR DESCRIPTION
**Pull request by ranjanan**
Added `complement` and `complement!`. Fixing tests now. Comments and suggestions welcome as I go along.  

Fixes #209 

cc: @kmsquire @phaverty @tkelman

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -2,6 +2,7 @@ using Base.Test
 using DataStructures
 const IntSet = DataStructures.IntSet
 import Compat: String
+using Primes
```
Comment (StefanKarpinski) [inline test/runtests.jl]: Where does this get used?

Comment (ranjanan) [inline test/runtests.jl]: The Primes package provides the `primes` function, which is deprecated in Base right?

File: test/test_circular_buffer.jl [hunk 2]
```diff
@@ -5,7 +5,8 @@ using Base.Test
 cb = CircularBuffer{Int}(5)
 @test length(cb) == 0
 @test capacity(cb) == 5
-@test_throws ArgumentError first(cb)
+# throws ArgumentError on v0.4 and BoundsError on v0.5
+@test_throws Exception first(cb)
```
Comment (tkelman) [inline test/test_circular_buffer.jl]: do you know what the version cutoff for this change was? might be better to be more specific

Comment (ranjanan) [inline test/test_circular_buffer.jl]: Yeah, I think https://github.com/JuliaLang/julia/pull/17137/ changed that behaviour

Comment (tkelman) [inline test/test_circular_buffer.jl]: run contrib/commit-name.sh on the merge commit https://github.com/JuliaLang/julia/commit/efa23adbbefca173290c3fafe35ebde8e7f8fe6a

Comment (tkelman) [inline test/test_circular_buffer.jl]: I meant in the test, not a comment - you should check for as specific an exception type as you can

**Comment by Keno** [comment]
LGTM.

**Comment by ranjanan** [comment]
@Keno @StefanKarpinski @tkelman can one of you guys tag a new release please?

**Comment by tkelman** [comment]
Sure.

**Pull request merged**